vulkan_helpers/vulkan_application.cpp

```diff
@@ -66,7 +67,7 @@ VulkanApplication::VulkanApplication(
     uint32_t device_peer_memory_size, bool use_protected_memory,
     bool use_host_query_reset, VkColorSpaceKHR swapchain_color_space,
     bool use_shared_presentation, bool use_mutable_swapchain_format,
-    const void* swapchain_extensions, bool use_vulkan_1_1, bool use_10bit_hdr,
+    const void* swapchain_extensions, uint32_t vulkan_api_version, bool use_10bit_hdr,
```

Comment:
I think changing this signature is dangerous. Any place using the old `VulkanApplication` constructor and passing a bool for `use_vulkan_1_1` will implicitly be convert that bool to a `uint32_t`. I found at least one place where this happens (application_sandbox/overlapping_frames/main.cpp).

Reply:
There is a CL that was a suggestion a while back for this (See my comment below) I 100% think thats the right thing to do, it might just take a bit of time. This constructor has gotten far too silly to make too many more changes to it if we can help it.

Reply:
I made a simplified version of that CL (#223). With that CL, I think we can leave this constructor as-is and just change `use_vulkan_1_1` to `vulkan_api_version` in `VulkanApplicationOptions`, and just use the `VulkanApplicationOptions` wherever we need `vulkan_api_version` (`SampleApplication` already uses it in that CL).

Reply:
Makes sense to me. We should probably also make sure any FUTURE tests always use this constructor instead of the old one. That way we can remove the old one.

Reply:
thanks for helping checking the cl. I didn't realize that there's alreayd one CL dealing with the option for the application. So close this PR and let using the #223 instead.

Reply:
#223 Does not add `vulkan_api_version`, it just changes the way arguments are passed to the `VulkanApplication` constructor. I was suggesting you rebase this change onto #223 (now merged), and change `use_vulkan_1_1` to `vulkan_api_version` in the `VulkanApplicationOptions` struct instead of the `VulkanApplication` constructor.